### DISCUSSION TITLE
Allow symlinks within a repository

### DISCUSSION THREAD

**Discussion by gg-kialo** [argoproj/argo-cd#9468] [Ideas]
The fix for GHSA-6gcg-hp2x-q54h (in this commit: https://github.com/argoproj/argo-cd/commit/d36d95dc9f71ec61c1a93794f81ece6d61a0d943), feels overly strict. It prevents symlinking to files that aren't in the directory root, but are in the same repository.

I would prefer if symlinks were only prevented from reaching outside of their own repository root.

I trust all the commits in my repo, and I make use of symlinks to deduplicate identical manifest files, i.e. a directory application contains yaml files, and a symlink to another directory of yaml files shared with other applications.

If this is still considered too insecure, then perhaps it could be a setting that could be enabled on the project or application level. I would be interested in contributing such changes if they are desirable.

If there are any other approaches to create an application of non-overlapping directories, I'd be eager to hear them.

**Comment by rumstead** [comment]
It sounds like you are describing [kustomize](https://github.com/kubernetes-sigs/kustomize).

Kustomize allows you to reference resources from different directories and even ones that are remote. 

1. Referencing bases - https://github.com/kubernetes-sigs/kustomize/tree/master/examples/multibases
2. Referencing remote bases - https://github.com/kubernetes-sigs/kustomize/blob/master/examples/remoteBuild.md

**Reply by gg-kialo** [reply]
Thanks for the tip. We've started using kustomize for this.

**Comment by crenshaw-dev** [comment]
You're quite right, symlinks within the repo should be allowed. I believe this was fixed with GHSA-jhqp-vf4w-rpwq.

There's even a [unit test](https://github.com/argoproj/argo-cd/blob/2ac7bd3039c7dd5f299b007221bcebaa3ef66fc1/reposerver/repository/repository_test.go#L2024-L2032) for this exact case. Here's the current bounds check: https://github.com/argoproj/argo-cd/blob/2ac7bd3039c7dd5f299b007221bcebaa3ef66fc1/reposerver/repository/repository.go#L1257
